qiskit/optimization/converters/linear_equality_to_penalty.py

```diff
@@ -103,12 +121,104 @@ def encode(self, op: QuadraticProgram, penalty_factor: float = 1e5,
+            terms.extend(coef for coef in constraint.linear.to_dict().values())
+        if any(isinstance(term, float) and not term.is_integer() for term in terms):
+            logger.warning(
+                'Warning: Using %f for the penalty coefficient because '
+                'a float coefficient exists in constraints. \n'
+                'The value could be too small. '
+                'If so, set the penalty coefficient manually.',
+                default_penalty,
+            )
+            return default_penalty
+
+        # (upper bound - lower bound) can be calculate as the sum of absolute value of coefficients
+        # Firstly, add 1 to guarantee that infeasible answers will be greater than upper bound.
+        penalties = [1.0]
+        # add linear terms of the object function.
+        penalties.extend(abs(coef) for coef in self._src.objective.linear.to_dict().values())
+        # add quadratic terms of the object function.
+        penalties.extend(abs(coef) for coef in self._src.objective.quadratic.to_dict().values())
+
+        return fsum(penalties)
+
+    def decode(self, result: OptimizationResult) -> OptimizationResult:
+        """Convert the result of the converted problem back to that of the original problem
+        Args:
+            result: The result of the converted problem.
+
+        Returns:
+            The result of the original problem.
+
+        Raises:
+            QiskitOptimizationError: if the number of variables in the result differs from
+                                     that of the original problem.
+        """
+        if len(result.x) != len(self._src.variables):
+            raise QiskitOptimizationError(
+                'The number of variables in the passed result differs from '
+                'that of the original problem.'
+            )
+        # Substitute variables to obtain the function value and feasibility in the original problem
+        substitute_dict = {}
+        variables = self._src.variables
+        for i in range(len(result.x)):
+            substitute_dict[variables[i].name] = result.x[i]
+        substituted_qp = self._src.substitute_variables(substitute_dict)
+
+        new_result = OptimizationResult()
+        new_result.x = result.x
+
+        # Set the new function value
+        new_result.fval = substituted_qp.objective.constant
+
+        # Set the new status of optimization result
+        if substituted_qp.status == QuadraticProgramStatus.VALID:
+            new_result.status = OptimizationResultStatus.SUCCESS
+        else:
+            new_result.status = OptimizationResultStatus.INFEASIBLE
+
+        return new_result
+
+    @property
+    def penalty(self) -> float:
+        """Returns the penalty factor used in conversion.
+
+        Returns:
+            The penalty factor used in conversion.
+        """
+        return self._penalty
+
+    @penalty.setter  # type:ignore
+    def penalty(self, penalty: float) -> None:
```

Comment:
Should this be `Optional[float]` the same as the constructor so that None can be set so it uses auto penalty value. And if so update docstring too to note this behavior